src/main/io/dataEdition.h

```diff
@@ -0,0 +1,220 @@
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * Authors:
+ * Dominic Clifton/Hydra
+ * Carsten Giesen
+ * Adam Majerczyk (majerczyk.adam@gmail.com)
+ * Texmode add-on by Michi (mamaretti32@gmail.com)
+ *
+ *  Created on: december 2015
+ *  Author: Pierre Archambault
+ */
+
+#define NO_COMMAND			0x00
+#define EXIT_EDITION_MODE	0x01	// THR_LO + YAW_CE + PIT_LO + ROL_LO
+#define ENTER_EDITION_MODE	0x02	// THR_LO + YAW_CE + PIT_LO + ROL_HI
+#define INC_COMMAND    		0x0B	// THR_LO + YAW_CE + PIT_HI + ROL_CE
+#define DEC_COMMAND			0x0D	// THR_LO + YAW_CE + PIT_LO + ROL_CE
+#define SET_COMMAND			0x09	// THR_LO + YAW_HI + PIT_HI + ROL_CE
+#define NEXT_COMMAND		0x0E	// THR_LO + YAW_CE + PIT_CE + ROL_HI
+#define PREV_COMMAND		0x07	// THR_LO + YAW_CE + PIT_CE + ROL_LO
+
+#define MILLISECONDS_IN_A_SECOND 	1000
+
+#define DEFAULT_DISPLAY_COLUMNS 	21
+#define DEFAULT_DISPLAY_LINES 		8
+
+
+typedef enum {
+    PIXEL_NORMAL = 0,
+    PIXEL_INVERTED = (1 << 0),
+    PIXEL_MIXED = (1 << 1)
+} pixelColor_e;
+
+#define LOWER_FLASH_SIZE_FOR_FULL_EDITION_MENUS    128
+#define DATA_EDITION_FULL_VERSION    (defined(FLASH_SIZE) && (FLASH_SIZE > LOWER_FLASH_SIZE_FOR_FULL_EDITION_MENUS))
+
+#if (!DATA_EDITION_FULL_VERSION)
+static const char * const pageNames[] = {
+	"All CleanF data"
+};
+typedef enum {
+	SELECTION_PAGE = 0,
+	ALL_DATA_PAGE
+} indexOfPages_e;
+// pageNames[] and indexOfPages must be kept synchronized
+
+static const char * const staticSharedLines[] = {
+    "CleanFlight page ?",
+    "        Save"
+};
+static const char * const staticDataPage[] = {
+    " ",
+    " index : %3i",
+    " %s",
+    " value : %9s",
+    "   min : %9d",
+    "   max : %9d"
+};
+
+#else
+
+static const char * const servoForwardChannel[] = {"CH1", "CH2", "CH3", "CH4", "A01",
+    "A02", "A03", "A04", "A05", "A06", "A07", "A08", "A09", "A10", "A11", "A12", " no"};
+
+static const char * const pageNames[] = {
+	"Firmware",
+    "All CleanF data",
+    "Features",
+    "Measures",
+    "Battery",
+    "Current_meter",
+    "ESC_Motors",
+    "PID_controller",
+    "PID_profile",
+    "Rate_profile",
+    "GPS",
+    "Receiver",
+    "Motors",
+    "Servos"
+};
+typedef enum {
+	SELECTION_PAGE = 0,
+	FIRMWARE_PAGE,
+    ALL_DATA_PAGE,
+    FEATURES_PAGE,
+	MEASURES_PAGE,
+    BATTERY_PAGE,
+    CURRENTMETER_PAGE,
+    ESCMOTORS_PAGE,
+    PIDCONTROLLER_PAGE,
+    PID_PROFILE_PAGE,
+    RATE_PROFILE_PAGE,
+    GPS_PAGE,
+    RECEIVER_PAGE,
+    MOTORS_PAGE,
+    SERVOS_PAGE
+} indexOfPages_e;
+// pageNames[] and indexOfPages must be kept synchronized
```

Comment:
you may use designated initializes to ensure this (https://gcc.gnu.org/onlinedocs/gcc/Designated-Inits.html)
```
pageNames = {
[FIRMWARE_PAGE] = "Firmware",
[ALL_DATA_PAGE] = "All CleanF data",
...
```
this will create implicit NULL at SELECTION_PAGE (which may simplify code anyway)